bootstrap_cfn/config.py

```diff
@@ -230,109 +441,161 @@ def elb(self):
-        template['BaseHostLaunchConfig']['Properties'][
-            'InstanceType'] = self.data['ec2']['parameters']['InstanceType']
-
-        # BLOCK DEVICE MAPPING
+        from troposphere import Ref, FindInMap, GetAZs, Base64, Join, awsencode
+        from troposphere.ec2 import SecurityGroup
+        from troposphere.autoscaling import LaunchConfiguration, \
+            AutoScalingGroup, BlockDeviceMapping, EBSBlockDevice, Tag
+
+        resources = []
+        sgs = []
+
+        for sg_name, ingress in self.data['ec2']['security_groups'].items():
+            sg = SecurityGroup(
+                sg_name,
+                VpcId=Ref("VPC"),
+                GroupDescription="BaseHost Security Group",
+                SecurityGroupIngress=ingress
+            )
+            sgs.append(sg)
+            resources.append(sg)
+
         devices = []
         try:
             for i in self.data['ec2']['block_devices']:
-                devices.append(
-                    {'DeviceName': i['DeviceName'], 'Ebs': {'VolumeSize': i['VolumeSize']}})
+                devices.append(BlockDeviceMapping(
+                    DeviceName=i['DeviceName'],
+                    Ebs=EBSBlockDevice(VolumeSize=i['VolumeSize']),
+                ))
         except KeyError:
-            devices.append(
-                {'DeviceName': '/dev/sda1', 'Ebs': {'VolumeSize': 20}})
-        template['BaseHostLaunchConfig']['Properties'][
-            'BlockDeviceMappings'] = devices
-
-        # SET AUTO SCALING PARAMETERS
-        template['ScalingGroup']['Properties'][
-            'MinSize'] = self.data['ec2']['auto_scaling']['min']
-        template['ScalingGroup']['Properties'][
-            'MaxSize'] = self.data['ec2']['auto_scaling']['max']
-        template['ScalingGroup']['Properties'][
-            'DesiredCapacity'] = self.data['ec2']['auto_scaling']['desired']
-
-        # SET INSTANCE TAGS
-        tags = []
-        for i in self.data['ec2']['tags']:
-            tags.append({'Key': i,
-                         'Value': self.data['ec2']['tags'][i],
-                         'PropagateAtLaunch': True})
-        template['ScalingGroup']['Properties']['Tags'] = tags
-
-        return template
+            devices.append(BlockDeviceMapping(
+                DeviceName="/dev/sda1",
+                Ebs=EBSBlockDevice(VolumeSize=20),
+            ))
+
+        launch_config = LaunchConfiguration(
+            "BaseHostLaunchConfig",
+            KeyName=self.data['ec2']['parameters']['KeyName'],
+            SecurityGroups=[Ref(g) for g in sgs],
+            InstanceType=self.data['ec2']['parameters']['InstanceType'],
+            AssociatePublicIpAddress=True,
+            IamInstanceProfile=Ref("InstanceProfile"),
+            ImageId=FindInMap("AWSRegion2AMI", Ref("AWS::Region"), "AMI"),
+            BlockDeviceMappings=devices,
+            UserData=Base64(Join("", [
+                "#!/bin/bash -xe\n",
+                "#do nothing for now",
```

Comment:
Might be nice to split this into a separate file (not a deal breaker)? I know we're not using it right now, but just so that it's not difficult later.